corehq/apps/linked_domain/updates.py

```diff
@@ -412,12 +461,12 @@ def update_user_roles(domain_link, overwrite=False):
 
 def _get_synced_role(upstream_role_def, downstream_domain, downstream_roles,
                      downstream_roles_by_name, ignore_conflicts):
-    role, conflicting_role = _get_matching_downstream_role(upstream_role_def,
+    role, conflicting_role, is_default_role = _get_matching_downstream_role(upstream_role_def,
```

Comment:
Yes. `is_default_role` was needed for this change because the spec wanted to report separate error messages for when 'default' roles had conflicts vs 'custom' ones. We felt the distinction was necessary because default/builtin roles that are created at project creation have changed over time. Because of that, starting with this ERM change, you could try syncing an upstream project to a downstream project that only had built in roles, which had never had problems syncing before -- and now failed to sync due to 'conflicts'. Basically, we felt users would be confused at conflicts from data they hadn't even touched.